models/media_ads/snowplow_media_player_media_ads.sql

```diff
@@ -0,0 +1,225 @@
+    max(a.duration_secs) as duration_secs,
+    sum(case when a.skippable then 1 else 0 end) > 0 as skippable,
+    avg(a.pod_position) as pod_position,
+
+    count(*) as views,
+    sum(case when a.clicked then 1 else 0 end) as clicked,
+    sum(case when a.skipped then 1 else 0 end) as skipped,
+    sum(case when a._25_percent_reached then 1 else 0 end) as _25_percent_reached,
+    sum(case when a._50_percent_reached then 1 else 0 end) as _50_percent_reached,
+    sum(case when a._75_percent_reached then 1 else 0 end) as _75_percent_reached,
+    sum(case when a._100_percent_reached then 1 else 0 end) as _100_percent_reached,
+
+    {% if is_incremental() %}
+    0 as views_unique,
+    0 as clicked_unique,
+    0 as skipped_unique,
+    0 as _25_percent_reached_unique,
+    0 as _50_percent_reached_unique,
+    0 as _75_percent_reached_unique,
+    0 as _100_percent_reached_unique,
+    {% else %}
+    count(distinct a.domain_userid) as views_unique,
+    count(distinct case when a.clicked then a.domain_userid end) as clicked_unique,
+    count(distinct case when a.skipped then a.domain_userid end) as skipped_unique,
+    count(distinct case when a._25_percent_reached then domain_userid end) as _25_percent_reached_unique,
+    count(distinct case when a._50_percent_reached then domain_userid end) as _50_percent_reached_unique,
+    count(distinct case when a._75_percent_reached then domain_userid end) as _75_percent_reached_unique,
+    count(distinct case when a._100_percent_reached then domain_userid end) as _100_percent_reached_unique,
+    {% endif %}
+
+    min(viewed_at) as first_view,
+    max(viewed_at) as last_view
+
+  from new_media_ad_views a
+
+  group by 1, 2, 3, 5
+
+)
+
+{% if is_incremental() %}
+
+, unique_counts_that_exist_in_new_data as (
+
+  select
+    a.media_ad_id,
+    count(distinct a.domain_userid) as views_unique,
+    count(distinct case when a.clicked then a.domain_userid end) as clicked_unique,
+    count(distinct case when a.skipped then a.domain_userid end) as skipped_unique,
+    count(distinct case when a._25_percent_reached then domain_userid end) as _25_percent_reached_unique,
+    count(distinct case when a._50_percent_reached then domain_userid end) as _50_percent_reached_unique,
+    count(distinct case when a._75_percent_reached then domain_userid end) as _75_percent_reached_unique,
+    count(distinct case when a._100_percent_reached then domain_userid end) as _100_percent_reached_unique
+
+  from {{ ref("snowplow_media_player_media_ad_views") }} a
+
+  where
+    -- enough time has passed since the page_view's start_tstamp to be able to process it as a whole (please bear in mind the late arriving data)
+    cast({{ dateadd('hour', var("snowplow__max_media_pv_window", 10), 'a.viewed_at') }} as {{ type_timestamp() }}) < {{ snowplow_utils.current_timestamp_in_utc() }}
+
+    -- exists in the new data
+    and a.media_ad_id in (select distinct b.media_ad_id from new_media_ad_views b)
+
```

Comment:
For large volumnes it is slightly more efficient to use an `exists` clause rather than an `in`